src/onepass.jl

```diff
@@ -73,64 +73,101 @@ parse!(p, ocp, e; log=false) = begin
+            :( ($names) ∈ R^$m, control  ) =>  p_control!(p, ocp, a, m; components_names=names, log)
+            :( ($names) ∈ R, $dummy      ) => return __throw("unknown syntax", p.lnum, p.line)
+            _                              => p_alias!(p, ocp, a, e1; log) # alias
+        end
+        # variable
+        :( ($e1, $e2) ∈ R^$q, variable ) => return __throw("unknown syntax: please provide a name to the variable", p.lnum, p.line)
+        :( $v ∈ R^$q, variable         ) => p_variable!(p, ocp, v, q; log)
+        :( $v ∈ R   , variable         ) => p_variable!(p, ocp, v   ; log)
+        :( $v       , variable         ) => p_variable!(p, ocp, v   ; log) # todo: remove
+        # time
+        :( $t ∈ [ $t0, $tf ], time     ) => p_time!(p, ocp, t, t0, tf; log)
+        # state
+        :( ($e1, $e2) ∈ R^$n, state    ) => return __throw("unknown syntax: please provide a name to the state", p.lnum, p.line)
+        :( $x ∈ R^$n, state            ) => p_state!(p, ocp, x, n; log)
+        :( $x ∈ R   , state            ) => p_state!(p, ocp, x   ; log)
+        :( $x       , state            ) => p_state!(p, ocp, x   ; log) # todo: remove
+        # control
+        :( ($e1, $e2) ∈ R^$m, control  ) => return __throw("unknown syntax: please provide a name to the control", p.lnum, p.line)
+        :( $u ∈ R^$m, control          ) => p_control!(p, ocp, u, m; log)
+        :( $u ∈ R   , control          ) => p_control!(p, ocp, u   ; log)
+        :( $u       , control          ) => p_control!(p, ocp, u   ; log) # todo: remove
+        # dynamics
+        :( ∂($x)($t) == $e1            ) => p_dynamics!(p, ocp, x, t, e1       ; log)
+        :( ∂($x)($t) == $e1, $label    ) => p_dynamics!(p, ocp, x, t, e1, label; log)
+        # constraints
+        :( $e1 == $e2                  ) => p_constraint!(p, ocp, e2     , e1, e2            ; log)
+        :( $e1 == $e2, $label          ) => p_constraint!(p, ocp, e2     , e1, e2,      label; log)
+        :( $e1 ≤  $e2 ≤  $e3           ) => p_constraint!(p, ocp, e1     , e2, e3            ; log)
+        :( $e1 ≤  $e2 ≤  $e3, $label   ) => p_constraint!(p, ocp, e1     , e2, e3     , label; log)
+        :(        $e2 ≤  $e3           ) => p_constraint!(p, ocp, nothing, e2, e3            ; log)
+        :(        $e2 ≤  $e3, $label   ) => p_constraint!(p, ocp, nothing, e2, e3     , label; log)
+        :( $e3 ≥  $e2 ≥  $e1           ) => p_constraint!(p, ocp, e1     , e2, e3            ; log)
+        :( $e3 ≥  $e2 ≥  $e1, $label   ) => p_constraint!(p, ocp, e1     , e2, e3     , label; log)
+        :( $e2 ≥  $e1                  ) => p_constraint!(p, ocp, e1     , e2, nothing       ; log)
+        :( $e2 ≥  $e1,        $label   ) => p_constraint!(p, ocp, e1     , e2, nothing, label; log)
+        # lagrange cost
+        :(             ∫($e1) → min    ) => p_lagrange!(p, ocp,     e1,         :min; log)
+        :(       $e1 * ∫($e2) → min    ) => p_lagrange!(p, ocp, :( $e1 * $e2 ), :min; log)
```

Comment:
@ocots convenient but: this kind of expression only makes sense if `e1` does not contain references to state or control (variable allowed, though). Otherwise we would allow something like
```julia
x(t) * ∫(u(t)) → min # nonsense
```
as equivalent to 
```julia
∫(x(t) * u(t)) → min
```
*etc.*

Reply:
Don't know if we have to check mathematical consistency.

Reply:
@ocots I am taking care of the changes suggested in the review above before merging.